### PR TITLE
feat: `HashType.from_composite_str` for Blake2b, SHA3, Blake3

### LINKED ISSUE #652: [BUG] `HashType.from_composite_str` fails to recognize `BLAKE2b`
In this file https://github.com/CycloneDX/cyclonedx-python-lib/blob/main/cyclonedx/model/__init__.py, line 440 cannot make the correct key in HashAlgorithm.
https://github.com/CycloneDX/cyclonedx-python-lib/blob/d7e517b160982fe9f4ea19d489f855db459d81fc/cyclonedx/model/__init__.py#L441-L444

### PR DESCRIPTION
**Pull request by schlenk**
The code mistreated hashes for Blake2b and SHA3.
Code for explicitly handling SHA1 & BLAKE3 was added, as those have no variants defined in the CycloneDX specification.

fixes #652

### CODE REVIEW

File: cyclonedx/model/__init__.py [hunk 1]
```diff
@@ -432,17 +437,32 @@ def from_composite_str(composite_hash: str) -> 'HashType':
...
+                alg=HashAlgorithm.SHA_1,
+                content=parts[1].lower()
+            )
         elif algorithm_prefix[0:3] == 'sha':
+            # This is actually SHA2...
             return HashType(
                 alg=getattr(HashAlgorithm, f'SHA_{algorithm_prefix[3:]}'),
                 content=parts[1].lower()
             )
-        elif algorithm_prefix[0:6] == 'blake2':
```
Comment (jkowalleck) [inline cyclonedx/model/__init__.py:440]: removing this case `blake2` might break existing downstream usage.
please add it back, below the new `blake2b` case.

**Review by jkowalleck — CHANGES_REQUESTED** [review]

File: cyclonedx/model/__init__.py (hunk 1 above)
Comment (schlenk) [reply cyclonedx/model/__init__.py:440]: Added it back, as it does no harm.

**Review by jkowalleck — APPROVED** [review]
thanks for the contribution, @schlenk

**Comment by jkowalleck** [comment]
released via https://github.com/CycloneDX/cyclonedx-python-lib/releases/tag/v7.6.0

**Pull request merged**